eclair-core/src/main/scala/fr/acinq/eclair/io/Switchboard.scala

```diff
@@ -34,7 +34,7 @@ class Switchboard(nodeParams: NodeParams, watcher: ActorRef, relayer: ActorRef,
   import Switchboard._
 
   // we load channels from database
-  {
+  val peersWithChannels: Set[PublicKey] = {
```

Comment:
True, but I don't see any other way (because we do need this information available until we reconnect all peers), and is it really an issue? With 1.000 distinct peers this would only be around 50KB of data.

Reply:
Actually I think I can avoid needing this map, let me dig into it a bit.

Reply:
This patch solves this: https://github.com/ACINQ/eclair/pull/1587#discussion_r569494216.